### PR TITLE
proxy: enable client certificate in transport

### PR DESCRIPTION
**Pull request by desimone**
This would enable downstream applications to enforce mutually authenticated connections for TLS connections.  An example app can be found in the [examples repository](https://github.com/pomerium/examples/tree/master/mutual-tls).

**Checklist**:
- [x] updated docs
- [x] unit tests added
- [x] related issues referenced
- [x] updated CHANGELOG.md
- [x] ready for review

### CODE REVIEW

File: internal/config/policy.go [hunk 1]
```diff
@@ -47,25 +49,45 @@ type Policy struct {
...
+		return fmt.Errorf("internal/config: policy bad destination url %s", err)
 	}
 
 	// Only allow public access if no other whitelists are in place
 	if p.AllowPublicUnauthenticatedAccess && (p.AllowedDomains != nil || p.AllowedGroups != nil || p.AllowedEmails != nil) {
-		return errors.New("internal/config: route marked as public but contains whitelists")
+		return fmt.Errorf("internal/config: policy route marked as public but contains whitelists")
+	}
+
+	if p.TLSClientCert != "" && p.TLSClientKey != "" {
```
Comment (travisgroth) [inline internal/config/policy.go]: Should probably throw an error if only one is configured.

**Comment by alagendijk-minddistrict** [comment]
Could you also add an option for these to be files instead of only the base64 encoded string?

**Comment by travisgroth** [comment]
>Could you also add an option for these to be files instead of only the base64 encoded string?

Agreed.  I read the cryptutil changes before sufficient coffee.

**Comment by desimone** [comment]
Thanks @alagendijk-minddistrict & @travisgroth ; 
- added support to add client certs as files in addition to as base64 blobs.
- added support to add custom ca (#179) as files too.  
- updated the example repo to prefer files which are easier to read
- added a few log statements for non-standard routes.

**Pull request merged**